services/profile/controller/controller.go

```diff
@@ -196,3 +199,57 @@ func GetValidFilteredProfiles(w http.ResponseWriter, r *http.Request) {
 
 	json.NewEncoder(w).Encode(filtered_profile_list)
 }
+
+/*
+	IsEventRedeemed checks the appropriate table to check whether the given event id has already been redeemed. If the event is not in the table, it add to the array.
+*/
+func IsEventRedeemed(w http.ResponseWriter, r *http.Request) {
+	id := r.Header.Get("HackIllinois-Identity")
+	event_id := mux.Vars(r)["id"]
+
+	redemption_status, err := service.EventRedemptionStatus(id, event_id)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Could not check if event was redeemed for id "+id+"."))
+		return
+	}
+	if !redemption_status {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Event with id "+id+" was already redeemed."))
+		return
+	}
+
+	json.NewEncoder(w).Encode(redemption_status)
+}
+
+/*
+	AwardPoints gives the specified number of points to the current user.
+*/
+func AwardPoints(w http.ResponseWriter, r *http.Request) {
+	id := r.Header.Get("HackIllinois-Identity")
+	points, err := strconv.Atoi(mux.Vars(r)["points"])
```

Comment:
This err is unused

Reply:
Again, reminder rhat this adds points to the wrong profile since only JWTs with `AdminRole` can hit this endpoint. A test to check for this would be nice.

Reply:
Refactored to not need this error (Using a post instead of a get)